pkg/controllers/manila/manila.go

```diff
@@ -123,6 +145,74 @@ func (c *ManilaController) sync(ctx context.Context, syncCtx factory.SyncContext
 	return c.setEnabledCondition(ctx)
 }
 
+func (c *ManilaController) getFsGroupPolicy(ctx context.Context) storagev1.FSGroupPolicy {
+
+	// Set the fsGroupPolicy param to the default value
+	fsGroupPolicy := "None"
+
+	// Try to get the current value defined for the operator
+	if c.isFSGroupPolicyAllowed(os.Getenv("CSI_FSGROUP_POLICY")) {
+		fsGroupPolicy = os.Getenv("CSI_FSGROUP_POLICY")
+	}
+
+	return storagev1.FSGroupPolicy(string(fsGroupPolicy))
+}
+
+func (c *ManilaController) applyCSIDriver(ctx context.Context, spec storagev1.CSIDriverSpec) error {
+	klog.V(4).Infof("Starting CSI driver config refresh")
+	defer klog.V(4).Infof("CSI driver config refresh finished")
+
+	var errs []error
+
+	crd, err := c.kubeClient.StorageV1().CSIDrivers().Get(ctx, "manila.csi.openstack.org", metav1.GetOptions{})
+	crd.Spec = spec
+
+	if err != nil {
+		crd = c.generateCSIDriver(spec)
```

Comment:
@jsafrane in L167 I get the existing CSIDriver object and patch it with the generated spec.
If an error occurs it means the object is not there, and I generate it with the existing function.

This shouldn't happen because as per your previous comment/review we want the operator to apply static resources when it starts, and `CSIDriver` object is one of them.

However, if I keep `csidriver.yaml` among the applied static resources [1], after updating the `CSISpec` map the operator reconciles it with the original object defined in the static file, which means we can't take advantage of [2] because after patching the object the `StaticResourceController` deletes and recreates it.

If I remove [1] and manually apply `csidriver.yaml` (or if I rely on the `generateCSIDriver` function), it works as expected.

Do you have a suggestion on how we can remove this `generateCSIDriver` function and apply the static resource defined in the `csidriver.yaml` file without making it part of the resources reconciled by the `StaticResourceController`?

[1] https://github.com/openshift/csi-driver-manila-operator/blob/master/pkg/operator/starter.go#L77
[2] https://github.com/openshift/library-go/pull/1324/commits/c4d0b28d0d214f590dd499961b1a55c209ef1822